### PR TITLE
feat: log auth0client param on authorize

### PR DESCRIPTION
**Pull request by danielgent**

### CODE REVIEW

File: src/routes/tsoa/authorize.ts [hunk 1]
```diff
@@ -101,6 +102,7 @@ export class AuthorizeController extends Controller {
     @Query("code_challenge_method") code_challenge_method?: CodeChallengeMethod,
     @Query("code_challenge") code_challenge?: string,
     @Query("realm") realm?: string,
+    @Query("auth0Client") auth0Client?: string,
```
Comment (danielgent) [inline src/routes/tsoa/authorize.ts:105]: I suppose this shouldn't be a required parameter

File: src/routes/tsoa/authorize.ts [hunk 2]
```diff
@@ -113,8 +115,15 @@ export class AuthorizeController extends Controller {
...
       throw new Error("Client not found");
     }
-    request.ctx.set("client_id", client.id);
     request.ctx.set("tenantId", client.tenant_id);
+    if (auth0Client) {
+      try {
+        const auth0ClientObj = JSON.parse(atob(auth0Client));
+        ctx.set("auth0_client", auth0ClientObj);
+      } catch (e) {
+        console.log("failed to parse auth0Client", e);
```
Comment (danielgent) [inline src/routes/tsoa/authorize.ts:124]: and we should not crash the route if this param is bad :man_shrugging:

File: src/types/Var.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ export type Var = {
   userId?: string;
   tenantId?: string;
   log: string;
-  logType: LogType;
+  logType?: LogType;
```
Comment (danielgent) [inline src/types/Var.ts:9]: this is correct. It's undefined initially

File: src/tsoa-middlewares/logger.ts [hunk 4]
```diff
@@ -268,6 +268,7 @@ export function loggerMiddleware(
       if (e instanceof HTTPException) {
         try {
           const logType = ctx.var.logType || logTypeInitial;
+          if (!logType) return e.getResponse();
```
Comment (danielgent) [inline src/tsoa-middlewares/logger.ts:271]: same thing as on a previous PR, if this isn't set, do not log.  This isn't datadog error monitoring, it's Auth0 logs reverse engineered

**Review by markusahlstrand — APPROVED** [review]
LGTM

**Pull request merged**